coordinator/src/services/spine-communication.ts

```diff
@@ -16,36 +30,128 @@ const httpsAgent = new https.Agent({
     ]
 });
 
-async function request(message: string) {
-    const wrappedMessage = addEbXmlWrapper(message)
-    try {
-        const result = await axios.post(
-            'https://veit07.devspineservices.nhs.uk/reliablemessaging/reliablerequest',
-            wrappedMessage,
-            {
-                httpsAgent,
-                headers: {
-                    "Content-Type": "multipart/related; boundary=\"--=_MIME-Boundary\"; type=text/xml; start=ebXMLHeader@spine.nhs.uk",
-                    "SOAPAction": "urn:nhs:names:services:mm/PORX_IN020101UK31"
+export class RequestHandler {
+
+    private spineEndpoint: string
+    private spinePath: string
+    private ebXMLBuilder: (message: string) => string
+
+    constructor(spineEndpoint: string, spinePath: string, ebXMLBuilder: (message: string) => string) {
+        this.spineEndpoint = spineEndpoint
+        this.spinePath = spinePath
+        this.ebXMLBuilder = ebXMLBuilder
+    }   
+
+    async request(message: string): Promise<SpineResponse> {
+        const wrappedMessage = this.ebXMLBuilder(message)
+
+        console.log(`Attempting to send the following message to spine:\n${wrappedMessage}`)
+
+        try {
+            const result = await axios.post<string>(
+                `${this.spineEndpoint}${this.spinePath}`,
+                wrappedMessage,
+                {
+                    httpsAgent,
+                    headers: {
+                        "Content-Type": "multipart/related; boundary=\"--=_MIME-Boundary\"; type=text/xml; start=ebXMLHeader@spine.nhs.uk",
+                        "SOAPAction": "urn:nhs:names:services:mm/PORX_IN020101SM31"
+                    }
                 }
-            },
-        )
-        return {body: result.data, statusCode: result.status}
-    } catch (error) {
-        if (error.response) {
-            return {body: error.response.data, statusCode: error.response.status}
-        } else if (error.request) {
-            return {body: error.request.data, statusCode: 408}
+            )
+
+            switch (result.status) {
+                case (202): {
+                    console.log('Successful post request for prescription message')
+                    const pollingUrl = result.headers['content-location']
```

Comment:
Does case matter here? I'd expect it to be Content-Location

Reply:
Hapi seems to normalise all the headers down to lower case. Its a 'feature': https://github.com/hapijs/hapi/issues/3273